code/modules/power/gravitygenerator.dm

```diff
@@ -392,6 +393,22 @@ GLOBAL_LIST_EMPTY(gravity_generators)
 
 // Misc
 
+/atom/movable/gravity_generator_warp_effect
```

Comment:
I feel like there's a lot of repeated code throughout here with so many of these defined, it'd be nice if there was some central place you could apply this? I think you were talking about putting it onto a component or element, but even just possibly making some base form of this with some arguments might be better.

Reply:
I could certainly make a base type then subtypes, but I wast sure about component or element, due to the effect that can be placed on it being so varied with possibilities, no real good way to do it with arguments. Open for advice.